homeassistant/components/netgear/config_flow.py

```diff
@@ -0,0 +1,199 @@
+"""Config flow to configure the Netgear integration."""
+from urllib.parse import urlparse
+
+from pynetgear import DEFAULT_HOST, DEFAULT_PORT, DEFAULT_USER
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_URL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+
+from .const import (  # pylint: disable=unused-import
+    CONF_CONSIDER_HOME,
+    CONF_METHOD_VERSION,
+    CONF_TRACKED_LIST,
+    DEFAULT_CONSIDER_HOME,
+    DEFAULT_METHOD_VERSION,
+    DOMAIN,
+)
+from .errors import CannotLoginException
+from .router import get_api
+
+
+def _discovery_schema_with_defaults(discovery_info):
+    return vol.Schema(_ordered_shared_schema(discovery_info))
+
+
+def _user_schema_with_defaults(user_input):
+    user_schema = {
+        vol.Optional(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+        vol.Optional(CONF_PORT, default=user_input.get(CONF_PORT, 0)): int,
+        vol.Optional(CONF_SSL, default=user_input.get(CONF_SSL, False)): bool,
+    }
+    user_schema.update(_ordered_shared_schema(user_input))
+
+    return vol.Schema(user_schema)
+
+
+def _ordered_shared_schema(schema_input):
+    return {
+        vol.Optional(CONF_USERNAME, default=schema_input.get(CONF_USERNAME, "")): str,
+        vol.Required(CONF_PASSWORD, default=schema_input.get(CONF_PASSWORD, "")): str,
+    }
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Init object."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        settings_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_METHOD_VERSION,
```

Comment:
Alternatively, if we can detect an unsupported method we should try one and then try the other one if the first one doesn't return ok data, and then remember that result.

Reply:
@MartinHjelmare I know little about the two diffrent update versions.
I know the current component uses V1 and that has been working fine for me (Netgear R7000).
I know the PR of @Quentame used V2 that I think is supposed to be more lightweigth on the router or something.
However V2 did not work that well for me because it listed devices unavailable while they were connected to my router, V1 did list them as available.

So both versions return data on an Netgear R7000, so can't select based on that.
However V2 did not work for all connected devices as it seemed.

I kind of don't want to dive into this V1/V2 since V1 worked fine for me and is the version currently used.

I can just remove the option to select it and always use V1 since that is what the current component uses?

Reply:
Yeah, I have a Netgear R6300v2 and V2 method doesn't work for it either. V1 works.

We need to decide on the method to use somehow that is compatible with known existing cases. The user should not need to care about this. If V1 works for all cases it's ok to stick with that. Maybe get in contact with the library maintainer(s) and ask if they know more?

Reply:
I think the best way is to just take V1 as the default, and then make a mapping of the router models to the V1/V2 method.
For now I know the Oribi models need V2 and we will get issues from people reporting that V1 does not work for them and we can add those models to V2.